### PR TITLE
Distinguish CI command runs

### PR DESCRIPTION
**Pull request by fotanus**
This patch add CI information on the user_agent in order for us to track
if a given command was run by a CI, as discussed in #3233.

### CODE REVIEW

File: spec/bundler/fetcher_spec.rb [hunk 1]
```diff
@@ -16,5 +16,21 @@
       expect(fetcher.user_agent).to match(/ruby\/(\d.)/)
       expect(fetcher.user_agent).to match(/options\/foo,bar/)
     end
+
+    describe "include CI information" do
+      it "from one CI" do
+        ENV["JENKINS_URL"] = "foo"
```
Comment (simi) [inline spec/bundler/fetcher_spec.rb]: this ENV should be cleaned

Comment (fotanus) [inline spec/bundler/fetcher_spec.rb]: Done! Thanks!

Comment (simi) [inline spec/bundler/fetcher_spec.rb]: It will be better to keep and reassign old value (instead of forcing nil).

Comment (fotanus) [inline spec/bundler/fetcher_spec.rb]: Of course! Silly me!

**Comment by fotanus** [comment]
Can I have this job on travis run again? I just tried locally and it worked with ruby 2.2.2. Looking at the travis log, it seems like a transient error. It have messages like this:

```
Could not fetch specs from file:/home/travis/build/bundler/bundler/tmp/gems/remote2/
```

**Comment by indirect** [comment]
restarted the build, I think it was failing because RubyGems master was broken

**Comment by fotanus** [comment]
@indirect Thank you!

**Comment by indirect** [comment]
Thanks!

**Pull request merged**